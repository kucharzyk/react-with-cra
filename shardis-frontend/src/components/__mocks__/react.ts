import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
