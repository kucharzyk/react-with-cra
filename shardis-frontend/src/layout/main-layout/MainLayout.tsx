import * as React from 'react';
import ServerTime from '../../server-time/ServerTime';
import AppHeader from '../app-header/AppHeader';
import Home from '../../home/Home';
import About from '../../about/About';

import { Route, Switch } from 'react-router-dom';
import NotFound from '../../not-found/NotFound';
import Paper from 'material-ui/Paper';

class MainLayout extends React.Component {
  render() {
    return (
      <div>
        <AppHeader/>
        <Paper>
          <Switch>
            <Route path={'/'} exact={true} component={Home}/>
            <Route path={'/home'} component={Home}/>
            <Route path={'/about'} component={About}/>
            <Route path={'/time'} component={ServerTime}/>
            <Route component={NotFound}/>
          </Switch>
        </Paper>
        <Paper style={{textAlign: 'center'}}>Shardis Â©2018</Paper>
      </div>
    );
  }
}

export default MainLayout;
