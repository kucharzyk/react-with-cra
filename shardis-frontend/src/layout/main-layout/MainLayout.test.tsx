import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import MainLayout from './MainLayout';
import { StaticRouter } from 'react-router-dom';

describe('MainLayout', () => {

  const LayoutWithRouter = () => (
    <StaticRouter location={'/'} context={{}}>
      <MainLayout/>
    </StaticRouter>
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LayoutWithRouter/>, div);
  });

  it('renders correctly', () => {
    const tree = create(<LayoutWithRouter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
