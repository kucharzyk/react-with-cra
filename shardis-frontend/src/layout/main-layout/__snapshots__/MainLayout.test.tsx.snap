// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders correctly 1`] = `
<div>
  <div
    className="App-header"
  >
    <header
      className="MuiPaper-root-8 MuiPaper-shadow4-14 MuiAppBar-root-1 MuiAppBar-positionStatic-4 MuiAppBar-colorDefault-5"
    >
      <div
        className="MuiToolbar-root-35 MuiToolbar-gutters-36"
      >
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />
        <h2
          className="MuiTypography-root-37 MuiTypography-title-43 MuiTypography-colorInherit-56"
          style={
            Object {
              "width": "50%",
            }
          }
        >
          <span
            className="app-title"
          >
            Welcome to React
          </span>
        </h2>
        <div
          className="MuiTabs-root-61"
        >
          <div
            className="MuiTabs-flexContainer-62"
          >
            <div
              className="MuiTabs-scrollingContainer-63 MuiTabs-fixed-64"
              onScroll={[Function]}
              role="tablist"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              <div
                className="MuiTabs-flexContainer-62"
              >
                <button
                  aria-selected={true}
                  className="MuiButtonBase-root-84 MuiTab-root-68 MuiTab-rootInherit-76 MuiTab-rootInheritSelected-77"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-80"
                  >
                    <div
                      className="MuiTab-labelContainer-81"
                    >
                      <span
                        className="MuiTab-label-82"
                      >
                        Main Page
                      </span>
                    </div>
                  </span>
                  <span
                    className="MuiTouchRipple-root-89"
                  />
                </button>
                <button
                  aria-selected={false}
                  className="MuiButtonBase-root-84 MuiTab-root-68 MuiTab-rootInherit-76"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-80"
                  >
                    <div
                      className="MuiTab-labelContainer-81"
                    >
                      <span
                        className="MuiTab-label-82"
                      >
                        Home
                      </span>
                    </div>
                  </span>
                  <span
                    className="MuiTouchRipple-root-89"
                  />
                </button>
                <button
                  aria-selected={false}
                  className="MuiButtonBase-root-84 MuiTab-root-68 MuiTab-rootInherit-76"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-80"
                  >
                    <div
                      className="MuiTab-labelContainer-81"
                    >
                      <span
                        className="MuiTab-label-82"
                      >
                        About
                      </span>
                    </div>
                  </span>
                  <span
                    className="MuiTouchRipple-root-89"
                  />
                </button>
                <button
                  aria-selected={false}
                  className="MuiButtonBase-root-84 MuiTab-root-68 MuiTab-rootInherit-76"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-80"
                  >
                    <div
                      className="MuiTab-labelContainer-81"
                    >
                      <span
                        className="MuiTab-label-82"
                      >
                        Time
                      </span>
                    </div>
                  </span>
                  <span
                    className="MuiTouchRipple-root-89"
                  />
                </button>
                <button
                  aria-selected={false}
                  className="MuiButtonBase-root-84 MuiTab-root-68 MuiTab-rootInherit-76"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-80"
                  >
                    <div
                      className="MuiTab-labelContainer-81"
                    >
                      <span
                        className="MuiTab-label-82"
                      >
                        404
                      </span>
                    </div>
                  </span>
                  <span
                    className="MuiTouchRipple-root-89"
                  />
                </button>
              </div>
              <div
                className="MuiTabIndicator-root-86 MuiTabIndicator-colorAccent-87"
                style={
                  Object {
                    "left": 0,
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiPaper-root-8 MuiPaper-shadow2-12 MuiPaper-rounded-9"
  >
    <div
      className="Home"
    >
      <h2>
        This is Home component
      </h2>
    </div>
  </div>
  <div
    className="MuiPaper-root-8 MuiPaper-shadow2-12 MuiPaper-rounded-9"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Shardis Â©2018
  </div>
</div>
`;
